apps/discovery_api/lib/discovery_api/event_handler.ex

```diff
@@ -62,6 +66,20 @@ defmodule DiscoveryApi.EventHandler do
     end
   end
 
+  def handle_event(%Brook.Event{type: dataset_delete(), data: %Dataset{} = dataset}) do
+    DiscoveryApi.RecommendationEngine.delete(dataset.id)
+    SystemNameCache.delete(dataset.technical.orgName, dataset.technical.dataName)
+    DiscoveryApi.Search.Storage.delete(dataset)
+    StatsCalculator.delete_completeness(dataset.id)
+    Model.delete(dataset.id)
+    ResponseCache.invalidate()
+    DataJsonService.delete_data_json()
+    Logger.debug("#{__MODULE__}: Deleted dataset for dataset: #{dataset.id}")
+  rescue
+    error ->
+      Logger.error("#{__MODULE__}: Failed to delete dataset for dataset: #{dataset.id}, Reason: #{inspect(error)}")
+  end
```

Comment:
Seems like both the happy and sad path should return `:discard`.  Right now they're both returning the result of the Logger function.